src/Layers/xrRenderPC_R4/r4_shaders.cpp

```diff
@@ -555,16 +556,16 @@ HRESULT CRender::shader_compile(pcstr name, IReader* fs, pcstr pFunctionName,
             const bool dx9compatibility = Flags & D3DCOMPILE_ENABLE_BACKWARDS_COMPATIBILITY;
             IWriter* file = FS.w_open(file_name);
 
+            file->w_u32(dx9compatibility);
             file->w_u32(fileCrc);
 
             u32 bytecodeCrc = crc32(pShaderBuf->GetBufferPointer(), pShaderBuf->GetBufferSize());
-            file->w_u32(bytecodeCrc);
-
-            file->w_u32(dx9compatibility);
+            file->w_u32(bytecodeCrc); // XXX: Do not write anything below this line, take a look at reading (crc32)
```

Comment:
XXX is a substitute for TODO, but there nothing to "todo".
```suggestion
            file->w_u32(bytecodeCrc); // Do not write anything below this line, take a look at reading (crc32)
```